include/util/json_renderer.hpp

```diff
@@ -64,131 +78,94 @@ struct Renderer
             }
             ++pos;
         }
-        out << buffer;
+        write(buffer);
     }
 
-    void operator()(const Object &object) const
+    void operator()(const Object &object)
     {
-        out << "{";
+        write('{');
         for (auto it = object.values.begin(), end = object.values.end(); it != end;)
         {
-            out << "\"" << it->first << "\":";
+            write('\"');
+            write(it->first);
+            write("\"");
             mapbox::util::apply_visitor(Renderer(out), it->second);
             if (++it != end)
             {
-                out << ",";
+                write(',');
             }
         }
-        out << "}";
+        write('}');
     }
 
-    void operator()(const Array &array) const
+    void operator()(const Array &array)
     {
-        out << "[";
+        write('[');
         for (auto it = array.values.cbegin(), end = array.values.cend(); it != end;)
         {
             mapbox::util::apply_visitor(Renderer(out), *it);
             if (++it != end)
             {
-                out << ",";
+                write(',');
             }
         }
-        out << "]";
+        write(']');
     }
 
-    void operator()(const True &) const { out << "true"; }
+    void operator()(const True &) { write("true"); }
 
-    void operator()(const False &) const { out << "false"; }
+    void operator()(const False &) { write("false"); }
 
-    void operator()(const Null &) const { out << "null"; }
+    void operator()(const Null &) { write("null"); }
 
   private:
-    std::ostream &out;
+    void write(const std::string &str);
+    void write(const char *str);
+    void write(char ch);
+
+  private:
+    Out &out;
 };
 
-struct ArrayRenderer
```

Comment:
# 1.

First of all I noticed that for some reason we use completely different implementation for `std::vector` and `std::ostream` outputs. I unified it using `std::ostream` implementation as common one.
<img width="497" alt="vector uses the same implementation" src="https://user-images.githubusercontent.com/266271/193300701-33e349a8-3615-40b7-a7f7-6097fd1b09ef.png">